app/models/transformation_mapping.rb

```diff
@@ -94,6 +95,8 @@ def validate_vm(vm, quick = true)
     # a valid vm must find all resources in the mapping and has never been migrated
     invalid_list = []
 
+    return VM_INACTIVE unless vm.active?
```

Comment:
Move this line into `#validate_v2v_migration` and move `vm.validate_v2v_migration` to the beginning of this method.

Reply:
@bzwei any suggestions for Aparna. We need this PR merged asap.

Reply:
```
def validate_vm(vm, quick = true)
  validate_error = vm.validate_v2v_migration
  return validate_error if validate_error

  invalid_list = []
  ...

  # remove vm.validate_v2v_migration
```

Reply:
Please try this:
```
def validate_vm(vm, quick = true)
  validate_result = vm.validate_v2v_migration
  return validate_result unless validate_result == VM_VALID

  invalid_list = []
  ...

  # remove vm.validate_v2v_migration
  VM_VALID
end
```

Reply:
Thanks @lfu - I can confirm that the above changes work.

Although, all this while, we have been trying hard not to directly use the `VM` constants like `VM_INACTIVE` ... and the above code uses `VM_VALID`. So, is that OK?

Reply:
@AparnaKarve ```VM_VALID``` is a contant in [TransformationMapping](https://github.com/ManageIQ/manageiq/blob/9734478c510996944ffeaf431298b8364a7cd568/app/models/transformation_mapping.rb#L7).

Reply:
`VM_INACTIVE` is also a new constant in `TransformationMapping` added in this PR

@lfu just trying to understand why this approach is preferable compared to the approach taken in the PR

Reply:
We are not trying to avoid using constants in this class. Logically `active?` check belongs to method `#validate_v2v_transformation`.
You will still need to add
```
  return TransformationMapping::VM_INACTIVE unless active?
```
in `Vm#validate_v2v_migration`.
Lucy has a refactoring PR to restructure the constants and methods. For now this is the solution.

Reply:
In the interest of time, I can just go ahead and make the above changes now
(although I am not sure how this approach is better than the other)

@bzwei Does the above code look OK? If yes, I can push an update shortly.
Thanks.

Reply:
New update pushed. deacfdc